sequence_task_1/student/include/impl.h

```diff
@@ -41,11 +55,71 @@ class StaticPlaylist
     void switchNext();
 
     /** @todo Amount of tracks in playlist */
-    size_type count() const;
+    size_type count() const { return (size_type)std::distance(icurrent, end()); }
 
     /** @todo Checks if playlist has any playable tracks */
-    bool hasTracks() const;
+    bool hasTracks() const { return (icurrent != end()); }
+
+    const_iterator begin() const { return m_tracklist.begin();  }
+    const_iterator end() const { return m_tracklist.end();  }
 
 private:
     Container m_tracklist;
 };
+
+
+
+template<class Container>
+template <class Container2>
+StaticPlaylist<Container>::StaticPlaylist(const Container2& track_list)
+    :
+      m_tracklist(track_list.begin(),track_list.end() )
+{
+    icurrent = m_tracklist.begin();
+}
+
+template<class Container>
+template <class Container2>
+StaticPlaylist<Container>& StaticPlaylist<Container>::operator= (const StaticPlaylist<Container2>& playList)
+{
+    if( &playList != this )
+    {
+        m_tracklist.assign(playList.begin(), playList.end());
+        icurrent = m_tracklist.begin();
+    }
+    return *this;
+}
+
+template<class Container>
+template< class... Args>
+const Song& StaticPlaylist<Container>::play(Args&&... songData)
+{
+    icurrent = m_tracklist.insert(m_tracklist.end(), Song{std::forward<Args>(songData)...} );
```

Comment:
Why do you create a temporary object if you have a construction data that is used by emplace?